src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -1972,9 +1972,35 @@ private void CompileAsVoidRemoveRethrow(Expression expr)
             CompileBlockEnd(end);
         }
 
+        private sealed class TryVisitor : ExpressionVisitor
+        {
+            public static readonly TryVisitor Instance;
+            
+            static TryVisitor()
+            {
+                Instance = new TryVisitor();
+            }
+
+            private static Expression ForceVoid(Expression node)
+            {
+                return node.Type == typeof(void) ? node : Expression.Block(typeof(void), node);
+            }
+
+            protected internal override Expression VisitTry(TryExpression node)
+            {
+                return node.Type != typeof(void) ? node : node.Update(ForceVoid(node.Body), Visit(node.Handlers, VisitCatchBlock), node.Finally, node.Fault);
+            }
+
+            protected override CatchBlock VisitCatchBlock(CatchBlock node)
```

Comment:
Finally shouldn't affect the stack after its been interpreted in any case, but I'll double check